lightning_client.go

```diff
@@ -2703,3 +2708,57 @@ func (s *lightningClient) NetworkInfo(ctx context.Context) (*NetworkInfo,
 		NumZombieChans:       resp.NumZombieChans,
 	}, nil
 }
+
+// InvoiceSubscription holds the parameters to specify from which update
+// to start streaming.
+type InvoiceSubscription struct {
+	//If specified (non-zero), then we'll first start by sending out
+	//notifications for all added indexes with an add_index greater than this
+	//value. This allows callers to catch up on any events they missed while they
+	//weren't connected to the streaming RPC.
+	AddIndex uint64
+	//If specified (non-zero), then we'll first start by sending out
+	//notifications for all settled indexes with an settle_index greater than
+	//this value. This allows callers to catch up on any events they missed while
+	//they weren't connected to the streaming RPC.
+	SettleIndex uint64
+}
+
+// SubscribeInvoices allows a client to subscribe to updates
+// of newly added/settled invoices.
+func (s *lightningClient) SubscribeInvoices(ctx context.Context,
+	req InvoiceSubscription) (<-chan *lnrpc.Invoice, <-chan error, error) {
+	rpcCtx := s.adminMac.WithMacaroonAuth(ctx)
+	invoiceStream, err := s.client.SubscribeInvoices(rpcCtx,
+		&lnrpc.InvoiceSubscription{
+			AddIndex:    req.AddIndex,
+			SettleIndex: req.SettleIndex,
+		})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	invoiceUpdates := make(chan *lnrpc.Invoice)
+	streamErr := make(chan error, 1)
+
+	// New invoices updates goroutine.
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+		for {
+			invoices, err := invoiceStream.Recv()
```

Comment:
nit: I actually meant the `invoices` variable, as it's just one. But the name `invoiceStream` is also better than plural.

Reply:
Fixed